File: src/Web.API/Features/GroupFeature/Queries/GetGroupByIdQuery/GetGroupByIdQueryResultDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Web.API.Features.GroupFeature.Queries.GetGroupByIdQuery;

public class GetGroupByIdQueryResultDTO
{
	public class Contact
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public string Email { get; set; }
		public string? ImageUrl { get; set; }
	}


	public int Id { get; set; }
	public string Name { get; set; }
	public string Description { get; set; }

	public DateTime CreatedAt { get; set; }
	public DateTime UpdatedAt { get; set; }
	public string UserId { get; set; }
	
	public List<Contact> Contacts { get; set; } = new List<Contact>();
}

File: src/Web.Angular/ClientApp/src/app/DTOs/Group/GetGroupByIdQueryResultDTO.ts
export class GetGroupByIdQueryResultDTO {

  public id: number;
  public name: string;
  public description: string;
  public contacts: { id: number, name: string, imageUrl: string }[];

  constructor()
  {
    this.id = 0;
    this.name = "";
    this.description = "";
    this.contacts = [];
  }
}
